src/Universe/UniverseGenerator.php

```diff
@@ -0,0 +1,601 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EtoA\Universe;
+
+use DirectoryIterator;
+use EtoA\Core\Configuration\ConfigurationService;
+use EtoA\Support\DatabaseManagerRepository;
+use EtoA\User\UserRepository;
+use Mutex;
+use UserToXml;
+
+/**
+ * @todo Create and use repositories for planets, stars etc.
+ */
+class UniverseGenerator
+{
+    private ConfigurationService $config;
+    private UserRepository $userRepo;
+    private SolarTypeRepository $solarTypes;
+    private PlanetTypeRepository $planetTypes;
+    private CellRepository $cellRepo;
+    private EntityRepository $entityRepo;
+    private StarRepository $starRepo;
+    private PlanetRepository $planetRepo;
+    private AsteroidsRepository $asteroidsRepo;
+    private NebulaRepository $nebulaRepo;
+    private WormholeRepository $wormholeRepo;
+    private EmptySpaceRepository $emptySpaceRepo;
+    private DatabaseManagerRepository $databaseManager;
+
+    /**
+     * @var array<int>
+     */
+    private array $sol_types = [];
+
+    /**
+     * @var array<int>
+     */
+    private array $planet_types = [];
+
+    public function __construct(
+        ConfigurationService $config,
+        UserRepository $userRepo,
+        SolarTypeRepository $solarTypes,
+        PlanetTypeRepository $planetTypes,
+        CellRepository $cellRepo,
+        EntityRepository $entityRepo,
+        StarRepository $starRepo,
+        PlanetRepository $planetRepo,
+        AsteroidsRepository $asteroidsRepo,
+        NebulaRepository $nebulaRepo,
+        WormholeRepository $wormholeRepo,
+        EmptySpaceRepository $emptySpaceRepo,
+        DatabaseManagerRepository $databaseManager
+    ) {
+        $this->config = $config;
+        $this->userRepo = $userRepo;
+        $this->solarTypes = $solarTypes;
+        $this->planetTypes = $planetTypes;
+        $this->cellRepo = $cellRepo;
+        $this->entityRepo = $entityRepo;
+        $this->starRepo = $starRepo;
+        $this->planetRepo = $planetRepo;
+        $this->asteroidsRepo = $asteroidsRepo;
+        $this->nebulaRepo = $nebulaRepo;
+        $this->wormholeRepo = $wormholeRepo;
+        $this->emptySpaceRepo = $emptySpaceRepo;
+        $this->databaseManager = $databaseManager;
+
+        $this->init();
+    }
+
+    private function init(): void
+    {
+        $this->sol_types = array_keys($this->solarTypes->getSolarTypeNames());
+        $this->planet_types = array_keys($this->planetTypes->getPlanetTypeNames());
+    }
+
+    public function exists(): bool
```

Comment:
Not sure what to think of those proxy methods. They don't really add much value.

Reply:
Fixed 4aa5fcc